src/Components/EnvironmentPicker/EnvironmentPickerManager.ts

```diff
@@ -114,7 +111,7 @@ export const getResourceUrls = (
     resourceType: AzureResourceTypes, // always pass this in case the resource is string type
     parentResource?: IAzureResource | string
 ) => {
-    return resources.map((resource) =>
+    return resources?.map((resource) =>
```

Comment:
why did we need to add the ?. Can't we just ensure we are passing valid data in? Ideally any state variables would initialize to [] instead of null.

Reply:
removed and made sure the array state variables are always set to empty array instead of null but also added default values to method params as guard: https://github.com/microsoft/iot-cardboard-js/pull/684/commits/357ab687abe3ed80be740fa4597775a1e50b1c2c#diff-9a856bccecc07dd21c6fb673efb9e1b33bce3bcbd41c9099d49cad741f8e958eR131